SEED-platform/seed
# Using selection tool for bsync validation

### Any background context?
This is the sister-pr to https://github.com/SEED-platform/seed/pull/2234

#### What's this PR do?
Adds an extra step to validate schema and use case when importing BuildingSync files
Also removes some vestigial front-end code for the old buildingsync import flow.

#### How should this be manually tested?
These tests should probably be run with and without celery workers, ie in local config include/exclude these lines
```bash
CELERY_TASK_ALWAYS_EAGER = True
CELERY_TASK_EAGER_PROPAGATES = True
```
When testing celery workers, it might be worth setting the log level to debug so messages can be seen: `celery ... -l debug`. Note that you WON'T see any messages when not running workers

Files for testing:
[use_case_data2.zip](https://github.com/SEED-platform/seed/files/4718996/use_case_data2.zip)

##### single valid file
- navigate to the data upload modal, click upload buildingsync
- select the file `buildingsync_v2_0_bricr_workflow_GOOD.xml`
- if looking at the network tab of debug tools, it hits `/validate_use_cases`
- eventually it should succeed and you should be allowed to go to the mapping page and continue like normal

##### single valid file with warnings
- navigate to the data upload modal, click upload buildingsync
- select this file: `buildingsync_v2_0_bricr_workflow_WARNINGS.xml`
- eventually it should SUCCEED and you should be allowed to go to the mapping page and continue like normal. You should ALSO see the warnings listed for the file

##### single file bad use case
- first, note the number of imported files for the data set you're going to upload to
- navigate to the data upload modal, click upload buildingsync
- select the file `ex_1_missing_required.xml`
- eventually it should say it failed to pass validation, and list use case errors and use case warnings, but no schema errors
- you should only be allowed to dismiss the dialog (not continue to mapping) and if you refresh the data page you should NOT see the number of files increase.

##### sing file bad schema
- first, note the number of imported files for the data set you're going to upload to
- navigate to the data upload modal, click upload buildingsync
- select the file `ex_1_bad_schema.xml`
- eventually it should say it failed to pass validation, and only show schema errors
- you should only be allowed to dismiss the dialog (not continue to mapping) and if you refresh the data page you should NOT see the number of files increase.

##### sing file that can't be parsed
- first, note the number of imported files for the data set you're going to upload to
- navigate to the data upload modal, click upload buildingsync
- select the file `ex_1_malformed.xml`
- eventually it should say it error out saying the request was unsuccessful (the file isn't parsable)
- you should only be allowed to dismiss the dialog (not continue to mapping) and if you refresh the data page you should NOT see the number of files increase.

##### zip file all good
- navigate to the data upload modal, click upload buildingsync
- select the file `many_all_good.zip`
- eventually it should succeed and you should be allowed to go to the mapping page and continue like normal

##### zip file one bad
- first, note the number of imported files for the data set you're going to upload to
- navigate to the data upload modal, click upload buildingsync
- select the file `many_one_bad.zip`
- eventually it should say it failed to pass validation, and only show use case warnings and errors for the bad file
- you should only be allowed to dismiss the dialog (not continue to mapping) and if you refresh the data page you should NOT see the number of files increase.

##### random api issues
These are some things to test if the validation api is down or we get an unexpected response.
1. test timeout by editing the timeout in `building_sync/validation_client.py` where the request is being made. Changing it to something like `1` should be sufficient (make sure you restart celery!). When uploading a buildingsync file it should fail saying it timed out
1. again in `validation_client.py` test a bad request by tweaking the API url, the default schema version or use case name (again remember to restart celery). These should all result in some sort of failure, and should be reported when trying to upload a buildingsync file

#### Screenshots (if appropriate)
Example failure
<img width="614" alt="Screen Shot 2020-05-29 at 4 50 41 PM" src="https://user-images.githubusercontent.com/18518728/83305821-71563a00-a1cf-11ea-858e-be37f1380b49.png">

Example success
<img width="612" alt="Screen Shot 2020-05-29 at 5 11 12 PM" src="https://user-images.githubusercontent.com/18518728/83305848-7a470b80-a1cf-11ea-83b4-d1773bf3b062.png">
